salt/_states/metalk8s_package_manager.py

```diff
@@ -0,0 +1,33 @@
+"""Custom state for managing yum packages and dependencies"""
+
+from __future__ import absolute_import
+
+import logging
+
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = "metalk8s_package_manager"
+
+
+def __virtualname__():
+    if __grains__['os_family'].lower() == 'redhat':
+        return __virtualname__
+    return (False, "metalk8s_package_manager: no yum based system detected")
```

Comment:
Let's not create artificial reasons: just state that this state only works on RedHat family of OSes.

Reply:
Or simply `RPM-based system`